### DISCUSSION TITLE
Searching page for text with ligatures not working

### DISCUSSION THREAD

**Discussion by reckoner** [pymupdf/PyMuPDF#1504] [Q&A]
**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is. Potentially add an issue reference.

I'm trying to do search for `text=modified` like this,

`page.search_for(text)`

but the text in the `page` has the word `modiﬁed` where the `fi` have been merged as a ligature so the search does not match.

**Describe the solution you'd like**

Search for text using correct encoding.

**Describe alternatives you've considered**
Are there several options for how your request could be met?
None

**Additional context**

**Comment by JorjMcKie** [comment]
Have you tried searching with switched off `fitz.TEXT_PRESERVE_LIGATURES`?

**Comment by reckoner** [comment]
If this is correct,

```python
page.search_for('modified',flags=fitz.TEXT_PRESERVE_LIGATURES)
```

then it is still not finding the `modified` text in the page that has the ligature.

**Comment by JorjMcKie** [comment]
you switched it ON not off!
Try flags=0 for a start.

**Comment by reckoner** [comment]
This works!

`page.search_for(text, flags=fitz.TEXT_ENCODING_LATIN)`

**Comment by JorjMcKie** [comment]
Yes, but only because `fitz.TEXT_ENCODING_LATIN == 0`.
Before a misconception establishes:
The flags parameter in text search and text extraction pertains to the underlying `TextPage` object. There is zero connection to the encoding of the underlying text.
The flags parameter **_only_** controls what becomes part of the `TextPage` and how:
* should ligatures be dissolved
* should general white space be transformed to normal space 0x20
* should images be extracted
* should spaces be generated between distant characters
* should spans always start a separate line
* should hyphenation dash be detected
* should text outside the mediabox be ignored

**Comment by JorjMcKie** [comment]
Look at the `TextPage` documentation to see which text function uses which flags to generate its `TextPage`.
If you use the `textage` parameter in these functions, then the flags used in generating that `TextPage` will prevail.

**Comment by reckoner** [comment]
OK. Let me study this.
